src/commands/init.ts

```diff
@@ -0,0 +1,111 @@
+import commander from "commander";
+import dotenv = require("dotenv");
+import * as fs from "fs";
+import yaml from "js-yaml";
+import * as os from "os";
+import { logger } from "../logger";
+
+export const defaultFileLocation = os.homedir() + "/.spk-config.yml";
+export let config: { [id: string]: any } = {};
+
+/**
+ * Adds the init command to the commander command object
+ * @param command Commander command object to decorate
+ */
+export const initCommandDecorator = (command: commander.Command): void => {
+  command
+    .command("init")
+    .alias("i")
+    .description("Initialize the spk tool for the first time")
+    .option("-f, --file <config-file-path>", "Path to the config file")
+    .action(async opts => {
+      try {
+        if (!opts.file) {
+          logger.error(
+            "You need to specify a file that stores configuration. "
+          );
+          return;
+        }
+        loadConfiguration(opts.file);
+        await writeConfigToDefaultLocation(opts.file);
+        logger.info("Successfully initialized the spk tool!");
+      } catch (err) {
+        logger.error(`Error occurred while initializing`);
+        logger.error(err);
+      }
+    });
+};
+
+/**
+ * Loads configuration from a given filename, if provided, otherwise
+ * uses the default file location ~/.spk-config.yml
+ * @param fileName file to load configuration from
+ */
+export const loadConfiguration = (fileName?: string) => {
+  if (!fileName) {
+    fileName = defaultFileLocation;
+  }
+  try {
+    fs.statSync(fileName);
+    dotenv.config();
+    const data = readYamlFile(fileName!);
+    loadConfigurationFromLocalEnv(data);
+  } catch (err) {
+    logger.error(`File ${fileName} does not exist.`);
+    throw err;
+  }
+};
+
+/**
+ * Loads configuration from local env
+ * @param configYaml configuration in object form
+ */
+export const loadConfigurationFromLocalEnv = (configObj: any) => {
```

Comment:
can you give more detail on what this is doing on the jsdoc?

Reply:
Done